### PR TITLE
Money counter

### PR DESCRIPTION
**Pull request by buggy213**
See #1075 

This PR adds a little textbox to the UI in order to display the amount of any given currency a player has.

ppmoney

### CODE REVIEW

File: ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -29,6 +29,7 @@ PlayerSettings:
   m_StackTraceTypes: 010000000100000001000000010000000100000001000000
```
Comment (hcorion) [inline ProjectSettings/ProjectSettings.asset]: Undo the changes to this file, we don't usually change these files.

Comment (buggy213) [inline ProjectSettings/ProjectSettings.asset]: huh. good point. I'm not exactly sure why that happened...

File: Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using UnityEngine.UI;
+
+public class CurrencyDisplay : MonoBehaviour
+{
+    public Text text;
+    
+    // Update is called once per frame
+    private void Update()
+    {
+        string[] currencies = World.Current.Wallet.GetCurrencyNames();
```
Comment (hcorion) [inline Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs]: You can cache this variable outside of Update and get it during Start, we don't need to get this every Update().

File: Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        StringBuilder text = new StringBuilder();
```
Comment (hcorion) [inline Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs]: You declare a public variable with the same name at the start, rename one of them.

File: Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        // Populate the text box
+        foreach (string currency in currencies)
+        {
+            text.Append(currency + ":");
+            text.Append(World.Current.Wallet[currency].Balance);
+            text.AppendLine();
+        }
+
+        // Don't question it ok.
+        this.text.text = text.ToString();
```
Comment (hcorion) [inline Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs]: This is extremely confusing, instead use the public variable you declared. assign it at the start using GetComponent, then you don't have to use that fancy this.text.text weirdness.

**Comment by hcorion** [comment]
Sorry, when I referenced that line in ProjectSettings/ProjectSettings.asset, I meant your changes to the entire file, not just that line.

**Comment by buggy213** [comment]
roight

**Comment by BraedonWooding** [comment]
Looks good, 😀.  Now I'm thinking maybe we should also have support for individual images instead of the text, but that would require specifying an image in the currencies xml/json file so I think a little out of scope?

**Comment by hcorion** [comment]
It's a good jumping off point.

**Comment by buggy213** [comment]
I think that's best left for a separate PR. This solution is functional and will work well enough right now, we can make it prettier later

**Comment by BraedonWooding** [comment]
Yeh that's what I thought but felt still valid to bring up 😀.

**Comment by buggy213** [comment]
It might be worth opening up an issue to request for make a request for artwork, although at this point it seems a little unnecessary.

**Comment by koosemose** [comment]
I could swear this used to be displayed on screen, but disappeared at some point...

Anyways, how does this interact with the "quest" overlay? i.e. does it display over top of it or under it (the preferable option

**Comment by buggy213** [comment]
Yeah, i made a PR for this a few months back but then i didn't have enough time to work on it and so it was closed. As of right now, the two currencies are entirely unobstructed by the quest overlay but even if we decide to add more, it does display under it.

**Comment by BraedonWooding** [comment]
Wait @buggy213 just realised this now it seems that the performance HUD (the frames, memory, max/min and average) is weird?  Normally they should sit in a 2x2 area rather than a 3x2.  So maybe just see if you edited it and if not it may be for some bizarre reasons and you can just make the prefab a little longer (width wise) :D.

**Comment by buggy213** [comment]
Huh. Alright, I'll do it tomorrow

**Comment by BraedonWooding** [comment]
Note: no pressure if for some reason you can't figure it out then we'll just ignore it and I can fix it once its merged into master :D.

**Comment by koosemose** [comment]
Let me know your progress on this, if you're going to be able to fix the issue @BraedonWooding pointed out or if you want to pass it on to him to fix.

**Comment by BraedonWooding** [comment]
I think @koosemose just merge it and when I get time today I'll go and fix the issue since it seems to actually be a 'big monitor' thing, and I will go and fix some/all of the UI for big monitor (depending if we want to do it by segments; i.e. Performance HUD UI fix, Dialog Box UI Fix...).

**Comment by bjubes** [comment]
@BraedonWooding  merging
